### PR TITLE
[hail/fs] addresses #6853

### PR DESCRIPTION
**Pull request by akotlar**
This puts a local filesystem option that is at parity with the remote option, and is triggered by the absence of a gs:// prefix.

A question: do we want to expose an rm?

### CODE REVIEW

**Comment by johnc1231** [comment]
Yes, I'll look today, sorry for delay.

File: hail/python/hail/fs/google_fs.py [hunk 1]
```diff
@@ -13,47 +15,111 @@ def __init__(self):
...
+                break
+            first_idx += 1
+
+        return "gs://" + path[first_idx:]
+
     def open(self, path: str, mode: str = 'r', buffer_size: int = 2**18):
+        if(self._is_local(path)):
+            if(not os.path.exists(path)):
+                parts = os.path.split(path)
+                # FIXME: Check compliance with HadoopFS
```
[the remark this answers is not in the document]
Comment (akotlar) [reply hail/python/hail/fs/google_fs.py]: Needed to check that the default behavior for our hadoop fs implementation was mkdir -p

[the remark this answers is not in the document]
Comment (johnc1231) [reply hail/python/hail/fs/google_fs.py]: Is it? Should we then delete this FIXME?

**Comment by akotlar** [comment]
@johnc1231 I updated this, after discovering that HadoopFS does mkdir -p,  but only during writes. I also made some small changes related to backend.py and ServiceBackend, to make it at least formally possible to use ServiceBackend (issue opened to this effect). Given the state of ServiceBackend and apiserver, I did not update test_google_fs_utils.py to explicitly check mkdir -p behavior, but this is implicitly checked in the local version of the file creation test.

File: hail/python/hail/context.py [hunk 2]
```diff
@@ -68,9 +68,9 @@ def __init__(self, sc=None, app_name="Hail", master=None, local='local[*]',
         jsc = sc._jsc.sc() if sc else None
 
         if _backend is None:
-            apiserver_url = os.environ.get('HAIL_APISERVER_URL')
-            if apiserver_url is not None:
-                _backend = ServiceBackend(apiserver_url)
+            #FIXME: either user HAIL_APISERVER_URL, or use another method to instantiate ServiceBackend
```
Comment (akotlar) [inline hail/python/hail/context.py]: https://github.com/hail-is/hail/issues/7068

**Comment by akotlar** [comment]
Bump

**Pull request merged**